### ISSUE TITLE
Give exception to user on pipeline `__init__()` when `stride` and `window_size` are incompatible

### ISSUE THREAD

**Issue by boyleconnor** [connor-qingxia/transformers#1]
### Feature request

This is tricky because we need to know in advance how many special tokens the tokenizer will add to each window. This may be impossible

@wigwit will experiment with figuring out how valid `max_length` and `stride` is validated when running the GPT tokenizer

### Motivation

It would be nice to give users an error message as early as possible

### Your contribution

@wigwit will work on this for now.

**Comment by wigwit** [comment]
I tested GPT2 tokenizer and how it would perform with the stride argument. There are two differences:
1. GPT2 does not have special tokens and padding tokens. Therefore, it does not make a difference for gpt2 whether to set adding_special_tokens =True or False.  This might incur inconsistency when averaging the logits since BERT has [CLS] and [SEP] at the beginning and end of the sentence.
2. GPT2 would require the user to explicitly assign a padding token since the tokenizer it self does not contain one. Other than that, the stride feature functions the same in GPT2 as other BERT based tokenizers.

**Comment by wigwit** [comment]
After reviewing the updated code I think (1) should not be a problem as we used `special_token_mask` to filter out the duplicated [CLS] and [SEP], and it should not mess with GPT2 in principle. But I will definitely double check on that with some test cases.
For (2), I am thinking if we could hardcode a pad token, or maybe there is workaround on that. Here are the list of special tokens for GPT2 and BERT for reference:
GP2:
```
            {'bos_token': '<|endoftext|>',
             'eos_token': '<|endoftext|>',
             'unk_token': '<|endoftext|>'}
````

BERT:
```
            {'unk_token': '[UNK]',
             'sep_token': '[SEP]',
             'pad_token': '[PAD]',
             'cls_token': '[CLS]',
             'mask_token': '[MASK]'}
```

**Comment by boyleconnor** [comment]
@wigwit So, let's say I want to make a BERT-based token classification pipeline:

```python
token_classifier = pipeline('token-classification-sliding-window', model='dslim/bert-base-NER', aggregation_strategy='FIRST', window_length=5, stride=3)
```

right now, our code lets me do this. But when I try to use that `token_classifier` I'll realize that this is an invalid combination of `window_length` and `stride`, because the BERT tokenizer inputs two special tokens for each window. Therefore if I run this:

```python
token_classifier("Hello, world, my name is Connor Boyle and I am working on this project with Qingxia Guo")
```

I get a big stacktrace ending with this error message:

```
PanicException: assertion failed: stride < max_len
```

I *want* to get a warning/error message telling me not to do this, already when I try to initialize a pipeline with a bad combination of `window_length` and `stride`.

**Comment by boyleconnor** [comment]
@wigwit Note that this varies per model, since GPT-based tokenizers (as you helpfully confirmed) don't seem to add any special tokens, so if I did the same thing with a GPT-based model:

```python
token_classifier = pipeline('token-classification-sliding-window', model='nguyenkhoa2407/gpt2-NER-favsbot', aggregation_strategy='FIRST', window_length=5, stride=3)
```

then I can actually run this just fine:

```python
token_classifier("Hello, world, my name is Connor Boyle and I am working on this project with Qingxia Guo")
```

Which outputs:

```python
[{'entity_group': 'LABEL_3', 'score': 0.3464192765350073, 'word': ' Hello,', 'start': 0, 'end': 6}, {'entity_group': 'LABEL_0', 'score': 0.40267269932127137, 'word': ' world, my name is', 'start': 6, 'end': 24}, {'entity_group': 'LABEL_3', 'score': 0.2950232108138331, 'word': ' Connor Boyle', 'start': 24, 'end': 37}, {'entity_group': 'LABEL_0', 'score': 0.4543407750912735, 'word': ' and I am working on this project with Qingxia', 'start': 37, 'end': 83}, {'entity_group': 'LABEL_4', 'score': 0.37418330750861495, 'word': ' Guo', 'start': 83, 'end': 87}]
```
